rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -554,6 +554,12 @@ impl Controller {
             Some(MAX_PARTITION_TIME),
             tokio::runtime::Handle::current(),
         )?;
+        connlib.set_dns(
+            callback_handler
+                .get_system_default_resolvers()
+                .expect("We expect to get the system's DNS")
+                .unwrap_or_default(),
```

Comment:
Same here, can we avoid calling our own callbacks and instead initialize it with the data that the callbacks return?